tempodb/backend/s3/s3.go

```diff
@@ -67,59 +70,27 @@ func (s *overrideSignatureVersion) IsExpired() bool {
 func New(cfg *Config) (backend.Reader, backend.Writer, backend.Compactor, error) {
 	l := log_util.Logger
 
-	wrapCredentialsProvider := func(p credentials.Provider) credentials.Provider {
-		if cfg.SignatureV2 {
-			return &overrideSignatureVersion{useV2: cfg.SignatureV2, upstream: p}
-		}
-		return p
-	}
-
-	creds := credentials.NewChainCredentials([]credentials.Provider{
-		wrapCredentialsProvider(&credentials.EnvAWS{}),
-		wrapCredentialsProvider(&credentials.Static{
-			Value: credentials.Value{
-				AccessKeyID:     cfg.AccessKey.String(),
-				SecretAccessKey: cfg.SecretKey.String(),
-			},
-		}),
-		wrapCredentialsProvider(&credentials.EnvMinio{}),
-		wrapCredentialsProvider(&credentials.FileAWSCredentials{}),
-		wrapCredentialsProvider(&credentials.FileMinioClient{}),
-		wrapCredentialsProvider(&credentials.IAM{
-			Client: &http.Client{
-				Transport: http.DefaultTransport,
-			},
-		}),
-	})
-
-	opts := &minio.Options{
-		Region: cfg.Region,
-		Secure: !cfg.Insecure,
-		Creds:  creds,
-	}
-
-	if cfg.ForcePathStyle {
-		opts.BucketLookup = minio.BucketLookupPath
+	core, err := createCore(cfg, false)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("unexpected error creating core: %w", err)
 	}
 
-	core, err := minio.NewCore(cfg.Endpoint, opts)
+	hedgedCore, err := createCore(cfg, true)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, fmt.Errorf("unexpected error creating hedgedCore: %w", err)
 	}
 
-	// TODO: add custom transport with instrumentation.
-	//client.SetCustomTransport(minio.DefaultTransport(!cfg.Insecure))
```

Comment:
Is this something we want to track in an issue? I noticed only GCS has instrumentation for now (which is not ideal, but probably fine).

Reply:
yup:

https://github.com/grafana/tempo/issues/758
https://github.com/grafana/tempo/issues/759